test/scaling/dimspecs.jl

```diff
@@ -0,0 +1,26 @@
+module ScalingDimspecTests
+
+using Interpolations, DualNumbers, Base.Test
+
+xs = -pi:(2pi/10):pi-2pi/10
+ys = -2:.1:2
+f(x,y) = sin(x) * y^2
+
+itp = interpolate(Float64[f(x,y) for x in xs, y in ys], Tuple{BSpline(Quadratic(Periodic)), BSpline(Linear)}, OnGrid)
+sitp = scale(itp, xs, ys)
+
+# Don't test too near the edges until issue #64 is resolved
+for (ix,x0) in enumerate(xs[5:end-5]), (iy,y0) in enumerate(ys[2:end-1])
+    x, y = x0 + 2pi/20, y0 + .05
+    @test_approx_eq sitp[x0, y0] f(x0,y0)
+    @test_approx_eq_eps sitp[x0, y0] f(x0,y0) 0.05
+
+    g = gradient(sitp, x, y)
+    fx = epsilon(f(dual(x,1), y))
+    fy = (f(x, ys[iy+2]) - f(x, ys[iy+1])) / (ys[iy+2] - ys[iy+1])
+
+    @test_approx_eq_eps g[1] fx 0.15
+    @test_approx_eq_eps g[2] fy 0.05 # gradients for linear interpolation is "easy"
```

Comment:
Consistency against the corresponding continuous function can of course only be expected to within relatively low precision, but we should insist on *internal* consistency of an interpolant to within machine precision. In other words, someone who uses finite-differencing or, better, DualNumbers to estimate the gradient of an interpolant should get the exact same answer we return from `gradient`---if that doesn't happen, then we've screwed up the gradient formula somehow.

For a 1d interpolant:
```jl
g = gradient(itp, x)
v = itp[dual(x,1)]
@test_approx_eq epsilon(v) g[1]
```
This is very different from (and much more important than) the tests above.